labelbox/data/serialization/labelbox_v1/converter.py

```diff
@@ -0,0 +1,93 @@
+from requests.exceptions import HTTPError
+import labelbox
+from typing import Any, Dict, Generator, Iterable
+import logging
+
+import ndjson
+import requests
+from google.api_core import retry
+
+from .label import LBV1Label
+from ...annotation_types.collection import (LabelCollection, LabelGenerator,
+                                            PrefetchGenerator)
+
+logger = logging.getLogger(__name__)
+
+
+class LBV1Converter:
+
+    @staticmethod
+    def deserialize_video(json_data: Iterable[Dict[str, Any]], client):
+        """
+        Converts a labelbox video export into the common labelbox format.
+
+        Args:
+            json_data: An iterable representing the labelbox video export.
+        Returns:
+            LabelGenerator containing the video data.
+        """
+        label_generator = (LBV1Label(**example).to_common()
+                           for example in LBV1VideoIterator(json_data, client))
+        return LabelGenerator(data=label_generator)
+
+    @staticmethod
+    def deserialize(json_data: Iterable[Dict[str, Any]]) -> LabelGenerator:
+        """
+        Converts a labelbox export (non-video) into the common labelbox format.
+
+        Args:
+            json_data: An iterable representing the labelbox export.
+        Returns:
+            LabelGenerator containing the export data.
+        """
+
+        def label_generator():
+            for example in json_data:
+                if 'frames' in example['Label']:
+                    raise ValueError(
+                        "Use `LBV1Converter.deserialize_video` to process video"
+                    )
+                yield LBV1Label(**example).to_common()
+
+        return LabelGenerator(data=label_generator())
+
+    @staticmethod
+    def serialize(
+            labels: LabelCollection) -> Generator[Dict[str, Any], None, None]:
+        """
+        Converts a labelbox common object to the labelbox json export format
+
+        Args:
+            labels: Either a LabelList or a LabelGenerator (LabelCollection)
+        Returns:
+            A generator for accessing the labelbox json export representation of the data
+        """
+        for label in labels:
+            res = LBV1Label.from_common(label)
+            yield res.dict(by_alias=True)
+
+
+class LBV1VideoIterator(PrefetchGenerator):
+    """
+    Generator that fetches video annotations in the background to be faster.
+    """
+
+    def __init__(self, examples, client):
+        self.client = client
+        super().__init__(examples)
+
+    def _process(self, value):
+        if 'frames' in value['Label']:
+            req = self._request(value)
+            value['Label'] = ndjson.loads(req)
```

Comment:
Is it necessary to mutate the original value, won't results be accessed via `__next__()`?

Reply:
We are actually returning the mutated payload:
https://github.com/Labelbox/labelbox-python/blob/6a71b2ab9f13536701b14b7472dce83d67e193f8/labelbox/data/serialization/labelbox_v1/converter.py#L82-L83. This generator is only updating part of the data.